erpnext/payroll/doctype/employee_incentive/employee_incentive.py

```diff
@@ -4,14 +4,23 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
 from frappe.model.document import Document
 
 class EmployeeIncentive(Document):
+	def validate(self):
```

Comment:
Currency should be validated on server-side. In case of data import or creationfrom API, client side code will not be executed. Hence, the currency will not be set.

It should be validated on server side that currency should match with salary structure.